.changeset/real-ducks-hope.md

```diff
@@ -0,0 +1,29 @@
+---
+"@latticexyz/cli": major
+"@latticexyz/world": major
+---
+
+The `registerFunctionSelector` function now accepts a single `functionSignature` string paramemer instead of separating function name and function arguments into separate parameters.
+
+```diff
+IBaseWorld {
+  function registerFunctionSelector(
+    ResourceId systemId,
+-   string memory systemFunctionName,
+-   string memory systemFunctionArguments
++   string memory systemFunctionSignature
```

Comment:
is it important to call this “system” here? I noticed the method doesn’t mention system

Reply:
I see now that there’s a difference between world and system function signatures.

Reply:
there is a nuance here that the naming is trying to make explicit: you're passing the signature of the function in the system contract (e.g. `increment()`), but the resulting world function signature will be `<namespace>_<systemName>_increment()`! That's why the returned selector is called `worldFunctionSelector`